src/app/services/impl/HttpExamQuestionSubmissionService.ts

```diff
@@ -33,7 +33,7 @@ export class HttpExamQuestionSubmissionService extends SubmissionService {
     super();
     this.httpRequestCache = new HttpRequestCache(http);
     this.baseUrl = baseUrl;
-    this.examId = 1; // TODO
+    this.examId = Number(/\/exams\/(?<examId>\d*)\//.exec(window.location.href).groups.examId);
```

Comment:
the heck/hack this is? lol


Give it an explanation as a comment above it stating why are you forced using regex